snap/integrity/dm_verity/veritysetup.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dm_verity
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type VeritySuperBlock struct {
+	Version        uint64
+	UUID           string
+	HashType       uint64
+	DataBlocks     uint64
+	DataBlockSize  uint64
+	HashBlocks     uint64
+	HashBlockSize  uint64
+	HashAlgorithm  string
+	Salt           string
+	HashDeviceSize uint64
+}
+
+func NewVeritySuperBlock() VeritySuperBlock {
+	sb := VeritySuperBlock{}
+	sb.Version = 1
+	return sb
+}
+
+func parseVeritySetupOutput(cmd *exec.Cmd) (string, *VeritySuperBlock, error) {
```

Comment:
Since this is just parsing the output, should it instead be passed the output string instead of the exec.Cmd?